static/js/admin.js

```diff
@@ -768,6 +794,66 @@ function _setup_page() {
         });
     });
 
+    $('.admin_filters_table').on('click', '.delete', function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var btn = $(this);
+
+        channel.del({
+            url: '/json/realm/filters/' + encodeURIComponent(btn.attr('data-filter-id')),
+            error: function (xhr, error_type) {
+                if (xhr.status.toString().charAt(0) === "4") {
+                    btn.closest("td").html(
+                        $("<p>").addClass("text-error").text($.parseJSON(xhr.responseText).msg)
+                    );
+                } else {
+                     btn.text("Failed!");
+                }
+            },
+            success: function () {
+                var row = btn.parents('tr');
+                row.remove();
+            }
+        });
+    });
+
+    $(".administration").on("submit", "form.admin-filter-form", function (e) {
+        e.preventDefault();
+        e.stopPropagation();
+        var filter_status = $('#admin-filter-status');
+        var pattern_status = $('#admin-filter-pattern-status');
+        var format_status = $('#admin-filter-format-status');
+        filter_status.hide();
+        pattern_status.hide();
+        format_status.hide();
+        var filter = {};
+        $(this).serializeArray().map(function (x){filter[x.name] = x.value;});
+
+        channel.put({
+            url: "/json/realm/filters",
+            data: $(this).serialize(),
```

Comment:
This code seems to have a bug where it doesn't strip whitespace properly; I accidentally had some trailing whitespace in some entries I created via the web UI and got the following filters (they don't work because of the `\t` characters):

```
$ ./manage.py realm_filters --r zulip.com 
zulip.com: [('#(?P<id>[0-9]+)\t', 'https://trac.example.com/ticket/%(id)s', 2), ('T(?P<id>[0-9]+)\t', 'https://trac.example.com/ticket/%(id)s', 3)]
```